### PR TITLE
Delete legacy payments code

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description
_Describe your changes and why you're making them. Please include the context, motivation, and relevant dependencies._

~Rebase on top of main after https://github.com/cal-itp/data-infra/pull/2833 is merged~

Resolves https://github.com/cal-itp/data-infra/issues/2836 (AC below)

- [x] Delete the unused Airflow code
- [ ]  Deprecate any old BigQuery tables following the [deprecation process](https://github.com/cal-itp/data-infra/blob/main/runbooks/data/deprecation-warehouse-models.md)
- [x]  Deprecate any unused dbt code
- [x]  Perhaps this is a good opportunity to fill out the [Payments section of the dbt overview docs page](https://github.com/cal-itp/data-infra/blob/main/warehouse/models/docs/_docs_overview.md)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

## How has this been tested?
_Include commands/logs/screenshots as relevant._

`poetry run dbt build -s +fct_payments_rides_v2+`

## Post-merge follow-ups
_Document any actions that must be taken post-merge to deploy or otherwise implement the changes in this PR (for example, running a full refresh of some incremental model in dbt). If these actions will take more than a few hours after the merge or if they will be completed by someone other than the PR author, please create a dedicated follow-up issue and link it here to track resolution._

- [ ] No action required
- [x] Actions required (specified below)
Deprecation process for legacy tables.

### CODE REVIEW

**Review by lauriemerrell — COMMENTED** [review]
Would you be willing to do the rest of https://github.com/cal-itp/data-infra/issues/2836 here?

**Comment by atvaccaro** [comment]
Unsure why this compile is failing, this model exists in prod.

File: warehouse/models/docs/_docs_overview.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The Cal-ITP Data Warehouse contains four core types of data (a few other data so
 
 * **Transit Database data, generally in `mart_transit_database`**: This is data hand-collected by Cal-ITP about transit in California, including things like organizations (whether they are transit providers, vendors, or other participants in the transit ecosystem); transit services; GTFS datasets; contracts; components of a transit technology stack; etc. For additional documentation of the Transit Database data model, see the [Airtable Data Documentation](https://docs.google.com/document/d/1KvlYRYB8cnyTOkT1Q0BbBmdQNguK_AMzhSV5ELXiZR4/edit#heading=h.u7y2eosf0i1d) or the older [Transit Database docs](https://docs.calitp.org/data-infra/datasets_and_tables/transitdatabase.html).
 
-* **Payments data, further docs to come after v2 migration**
+* **Payments data**: Data provided by fare processors (such as Littlepay) that includes device transactions, payments, funding sources, etc. Currently this data is sourced from vendor-provided published artifacts (such as S3 files) and used to power agency-facing dashboards showing usage of contactless payments. Mart models for this data must have row-level access controls to ensure agencies only see their data.
```
Comment (lauriemerrell) [inline warehouse/models/docs/_docs_overview.md:12]: can you list what mart schema this is in following the others

**Review by lauriemerrell — APPROVED** [review]
just one docs nit

**Comment by atvaccaro** [comment]
Merging sincE I think it'll be OK in prod

**Pull request merged**